integration/ztoc_test.go

```diff
@@ -100,3 +93,11 @@ func TestSociZtocList(t *testing.T) {
 		}
 	})
 }
+
+func sociIndexFromDigest(t *testing.T, sh *shell.Shell, indexDigest string) (index soci.Index) {
```

Comment:
Maybe return an error here and do a `t.Fatal` in the test? I don't know how I feel about non-test functions being able to fail the test.

Reply:
changed both `sociIndexFromDigest` and `buildIndexByManipulatingZtocData` to not require a `t *tesing.T` but return an error.

Reply:
also move `sociIndexFromDigest` to `index_test.go`, since it's more related to `index` and used by multiple test files. (probably we should have some `testutil_index.go, testutil_ztoc.go` etc, but before we have that it might be good to collocate these utils with most relevant tests)